hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1919,7 +1920,7 @@ public HoodieRecord next() {
   public static Schema getProjectedSchemaForFunctionalIndex(HoodieFunctionalIndexDefinition indexDefinition, HoodieTableMetaClient metaClient) throws Exception {
     TableSchemaResolver schemaResolver = new TableSchemaResolver(metaClient);
     Schema tableSchema = schemaResolver.getTableAvroSchema();
-    return HoodieAvroUtils.getSchemaForFields(tableSchema, indexDefinition.getSourceFields());
+    return addMetadataFields(getSchemaForFields(tableSchema, indexDefinition.getSourceFields()));
```

Comment:
Is this change related with this fix?

Reply:
yeah for update we need record key field.